packages/dd-trace/src/format.js

```diff
@@ -30,11 +30,13 @@ const map = {
   'resource.name': 'resource'
 }
 
-function format (span) {
+function format (span, exporter) {
+  const encoder = exporter?._writer?._encoder
+
   const formatted = formatSpan(span)
 
   extractSpanLinks(formatted, span)
-  extractSpanEvents(formatted, span)
+  extractSpanEvents(formatted, span, encoder, exporter?._agentSupportsTopLevelSpanEvents)
```

Comment:
Since this value is only ever read from the outside should we really prefix it with an underscore as that signifies it's a private value?